README.md

```diff
@@ -94,6 +106,43 @@ This will place four binaries in your `$GOBIN`;
 
 Make sure that your `$GOBIN` is in your `$PATH`.
 
+You can delete the `tools` folder as you don't need it any more.
+
+We need a new project with go modules initialized so go will autodetect the GO111MODULES support. The name of the project does not matter.
+
+`go mod tidy` will pull in the files for each of these tools into `$GOPATH/src` as regular modules.
+
+`go generate tools.go` will run the `//go:generate` command in the file itself which will install the binaries for these tools based on the source files that `go mod tidy` pulled in.
+
+You can also add these imports into an existing project and generate them as tools by using the `// +build tools` flag, for example:
+
+```go
+// +build tools
+
+package tools
+
+import (
+    _ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
+    _ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-openapiv2"
+    _ "google.golang.org/grpc/cmd/protoc-gen-go-grpc"
+    _ "google.golang.org/protobuf/cmd/protoc-gen-go"
+)
+```
+
+For more information, see documentation on [Go Modules](https://github.com/golang/go/wiki/Modules), [tool dependencies](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module).
+
+Please note that you should be in `GO111MODULES=auto` or `on` modes as `GOPATH` support is planned to be dropped in Go 1.17. See [the changes to go modules in 1.16](https://blog.golang.org/go116-module-changes) .
+
+If you'd rather not use the `go:generate` comment, you can install the tools manually by running the following command after `go mod tidy`:
+
+```sh
+$ go install \
+    github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway \
+    github.com/grpc-ecosystem/grpc-gateway/protoc-gen-openapiv2 \
+    google.golang.org/protobuf/cmd/protoc-gen-go \
+    google.golang.org/grpc/cmd/protoc-gen-go-grpc
```

Comment:
With go1.16.4 I get:
```
$ export GO111MODULES=on
$ go install \
>     github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
go install: version is required when current directory is not in a module
	Try 'go install github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway@latest' to install the latest version
```

The suggested `@latest` addition fixes the installation.